android-json-form-wizard/src/main/java/com/vijay/jsonwizard/task/AttachRepeatingGroupTask.java

```diff
@@ -102,20 +102,22 @@ protected void onPostExecute(List<View> result) {
                     keysToRemove.add(repeatingGroupKey);
                     rootLayout.removeViewAt(i);
                 }
-//                remove deleted fields from form json
-                ArrayList<String> removeThisFields = new ArrayList<>();
+                //remove deleted fields from form json
+                ArrayList<String> deletedFields = new ArrayList<>();
                 int len = fields.length();
                 for (int i = len - 1; i >= 0; i--) {
                     String[] key = ((String) fields.getJSONObject(i).get(KEY)).split("_");
                     if (keysToRemove.contains(key[key.length - 1])) {
```

Comment:
Since you are now adding fields to the `formFields` map, can this loop be added to the loop above in line 100 so that the lookup is faster via the map without iterating all the fields